seismicpro/src/utils/plot_utils.py

```diff
@@ -15,19 +15,17 @@ def save_figure(fig, fname, dpi=100, bbox_inches="tight", pad_inches=0.1, **kwar
     """Save the given figure. All `args` and `kwargs` are passed directly into `matplotlib.pyplot.savefig`."""
     fig.savefig(fname, dpi=dpi, bbox_inches=bbox_inches, pad_inches=pad_inches, **kwargs)
 
-
-def set_text_formatting(kwargs):
-    """Pop text formatting args from `kwargs` and set them as defaults for 'title', 'x_ticker' and 'y_ticker'."""
+def set_text_formatting(*args, **kwargs):
+    """Pop text formatting args from `kwargs` and set them as defaults for args transformed to dickts."""
     FORMAT_ARGS = {'fontsize', 'size', 'fontfamily', 'family', 'fontweight', 'weight'}
-    TEXT_ARGS = {'title', 'x_ticker', 'y_ticker'}
 
     global_formatting = {arg: kwargs.pop(arg) for arg in FORMAT_ARGS if arg in kwargs}
-    text_args = {arg: {**global_formatting, **as_dict(kwargs.pop(arg), key="label")}
-                 for arg in TEXT_ARGS if arg in kwargs}
-    return {**kwargs, **text_args}
-
+    text_args = ({**global_formatting, **({} if arg is None else as_dict(arg, key="label"))}
+                  for arg in args)
```

Comment:
I'd probably split it into two operations: first cast each `arg` to a dict and then merge it with defaults

Reply:
Like this? I don't fancy it, but would switch to this one If others agree

```suggestion
    dict_args = ({} if arg is None else as_dict(arg, key="label") for arg in args)
    text_args = ({**global_formatting, **arg} for arg in dict_args)
```